sympy/printing/tests/test_tree.py

```diff
@@ -2,178 +2,13 @@
-        '|   complex: True\n',
-        '|   extended_negative: False\n',
-        '|   extended_nonnegative: True\n',
-        '|   extended_real: True\n',
-        '|   finite: True\n',
-        '|   hermitian: True\n',
-        '|   imaginary: False\n',
-        '|   infinite: False\n',
-        '|   integer: True\n',
-        '|   irrational: False\n',
-        '|   negative: False\n',
-        '|   noninteger: False\n',
-        '|   nonnegative: True\n',
-        '|   rational: True\n',
-        '|   real: True\n',
-        '|   transcendental: False\n',
-        '+-MatrixSymbol: B\n',
-        '  algebraic: False\n',
-        '  commutative: False\n',
-        '  complex: False\n',
-        '  composite: False\n',
-        '  even: False\n',
-        '  extended_negative: False\n',
-        '  extended_nonnegative: False\n',
-        '  extended_nonpositive: False\n',
-        '  extended_nonzero: False\n',
-        '  extended_positive: False\n',
-        '  extended_real: False\n',
-        '  imaginary: False\n',
-        '  integer: False\n',
-        '  irrational: False\n',
-        '  negative: False\n',
-        '  noninteger: False\n',
-        '  nonnegative: False\n',
-        '  nonpositive: False\n',
-        '  nonzero: False\n',
-        '  odd: False\n',
-        '  positive: False\n',
-        '  prime: False\n',
-        '  rational: False\n',
-        '  real: False\n',
-        '  transcendental: False\n',
-        '  zero: False\n',
-        '  +-Symbol: B\n',
-        '  | commutative: True\n',
-        '  +-Integer: 3\n',
-        '  | algebraic: True\n',
-        '  | commutative: True\n',
-        '  | complex: True\n',
-        '  | extended_negative: False\n',
-        '  | extended_nonnegative: True\n',
-        '  | extended_real: True\n',
-        '  | finite: True\n',
-        '  | hermitian: True\n',
-        '  | imaginary: False\n',
-        '  | infinite: False\n',
-        '  | integer: True\n',
-        '  | irrational: False\n',
-        '  | negative: False\n',
-        '  | noninteger: False\n',
-        '  | nonnegative: True\n',
-        '  | rational: True\n',
-        '  | real: True\n',
-        '  | transcendental: False\n',
-        '  +-Integer: 3\n',
-        '    algebraic: True\n',
-        '    commutative: True\n',
-        '    complex: True\n',
-        '    extended_negative: False\n',
-        '    extended_nonnegative: True\n',
-        '    extended_real: True\n',
-        '    finite: True\n',
-        '    hermitian: True\n',
-        '    imaginary: False\n',
-        '    infinite: False\n',
-        '    integer: True\n',
-        '    irrational: False\n',
-        '    negative: False\n',
-        '    noninteger: False\n',
-        '    nonnegative: True\n',
-        '    rational: True\n',
-        '    real: True\n',
-        '    transcendental: False\n'
-    ]
-
-    assert tree(A + B) == "".join(test_str)
+    # XXX After making the _assumptions cache deterministic,
```

Comment:
Why did this file change? This seems unrelated to `MatrixSymbol`

Reply:
This uses matrix symbols for test in printing.
And I’d find it painful to change every assumptions cache and its order even if they are generated in randomized way

Reply:
But why do it in this PR? The test is already marked `XFAIL`.

Reply:
I wanted to make that this test should be run with dry run.

Reply:
Or this can just be changed to test against actual string output if it doesn’t print out messy assumptions again

Reply:
I clearly see that there is no way to avoid printing all the unnecessary assumptions

Reply:
Is the issue that this prints lots of assumptions about `Str` itself, which obviously mean nothing?

Reply:
Yes, and I just think that it’s the flaw of the original design of the printer

Reply:
That sounds like a flaw in `Str` to me - assuming that a string is real / commutative etc doesn't make any sense. Marking as unresolved in case @asmeurer has thoughts.

Reply:
It isn't only a problem of Str having assumptions, but all other sympy objects because Basic inherits assumptions from Expr.
Are you just saying you don't like the design of Str?

Reply:
I still think we shouldn't have Str in the first place, but if we do have it, it makes sense to me that it should be a Basic, not Expr object.

Reply:
I guess it already is Basic only, but the problem is that the default assumptions live on Basic rather than Expr. I think @oscarbenjamin discussed this issue before. It probably won't be easy to fix, but it is unrelated to Str specifically.

Reply:
There are two problems here:

1. The old assumptions are applied to all of Basic rather than just Expr. I have a PR #19474 which experiments with changing that. It would not be hard to finish off that PR. The only thing missing is a deprecation warning wrapper that allows the old assumptions to keep working on Basic but with a warning.

2. The `tree` function should not print assumptions for all objects but rather only for those objects for which the assumptions are part of the definition (`Symbol`, `AppliedUndef`, anything else?). It should also only print those assumptions that are part of that definition (`.assumptions0`).